spring-transaction/src/test/kotlin/org/jetbrains/exposed/spring/ExposedTransactionManagerTest.kt

```diff
@@ -87,45 +96,162 @@ open class ExposedTransactionManagerTest : SpringTransactionTestBase() {
     @Commit
     @Transactional
     open fun testConnectionCombineWithExposedTransaction2() {
-        SchemaUtils.create(T1)
         val rnd = Random().nextInt().toString()
         T1.insert {
             it[c1] = rnd
         }
-
         Assert.assertEquals(T1.selectAll().single()[T1.c1], rnd)
 
         transaction {
-            val rnd = Random().nextInt().toString()
-            T1.insert {
-                it[c1] = rnd
-            }
+            T1.insertRandom()
+            Assert.assertEquals(2, T1.selectAll().count())
+        }
+    }
 
+    @Test
+    @Repeat(5)
+    @Transactional
+    open fun testConnectionWithNestedTransactionCommit() {
```

Comment:
Please add a simple comment about the expected behavior of the different propagation levels to each relevant test (helps future maintainers that might not know the specifics of each level).

For example, I think using the one-line description for each level from the 'enum constant summary' section of the [spring reference docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/transaction/annotation/Propagation.html) would be helpful and enough.

Since REQUIRED is default, don't worry about adding a comment for it to any of the above tests.

Reply:
Btw adding these comments in `ExposedTransactionManagerTest.kt` should be enough. I don't think you need to duplicate the comments in the tests in `SpringTransactionManagerTest.kt`, but I'll leave that choice up to you.

Reply:
I've added comments to the test, and in the case of failure cases, I've added a reason as to why it's failing.